app/src/main/java/io/github/sheasmith/discountdetective/dao/ShoppingListDao.kt

```diff
@@ -28,4 +28,21 @@ interface ShoppingListDao {
      */
     @Delete
     suspend fun delete(shoppingListItem: ShoppingListItem)
+
+    /**
+     * Update the checked state of an item
+     */
+    @Query(
+        "UPDATE shopping_list SET " +
+                "checked = :checked WHERE " +
+                "productId = :productId AND " +
+                "retailerProductInformationId = :retailerProductInformationId AND " +
+                "storeId = :storeId"
+    )
+    suspend fun updateChecked(
```

Comment:
Maybe this should be a generic Room @Update? https://developer.android.com/reference/androidx/room/Update

That way, we don't need to add a new method every time we want to update some other parameter? Especially since the repo has the shopping list item anyway.

Reply:
Yup, that's a nice method